app/src/main/java/com/orgzly/android/util/OrgFormatter.kt

```diff
@@ -382,6 +385,143 @@ object OrgFormatter {
         return ":$LOGBOOK_DRAWER_NAME:\n$entry\n:END:$prefixedContent"
     }
 
+    @JvmStatic
+    fun getLastClockInPosition(content: String?): ArrayList<Int> {
+        val m = LOGBOOK_DRAWER_PATTERN.matcher(content)
+
+        val pos = ArrayList<Int>()
+
+        if (m.find()) {
+            val start = m.start(2) // Content start
+            val end = m.end()
+
+            val CLOCKED_TIMES_P = Pattern.compile("(CLOCK: *" + FULL_TIMESTAMP + ") *(-- *" + FULL_TIMESTAMP + ")?( *=> *[0-9]{1,4}:[0-9]{2})?[\\r\\n]*")
```

Comment:
I believe you can reference variables using the `$` symbol to build the `CLOCKED_TIMES_P` regex string
```suggestion
            val CLOCKED_TIMES_P = Pattern.compile("(CLOCK: *$FULL_TIMESTAMP) *(-- *$FULL_TIMESTAMP)?( *=> *[0-9]{1,4}:[0-9]{2})?[\\r\\n]*")
```

Reply:
Thanks, the change has been implemented in the last commit !